geopy/geocoders/googleplaces.py

```diff
@@ -0,0 +1,217 @@
+"""
+:class:`.GooglePlaces` is the Google Places API.
+"""
+from geopy.compat import urlencode
+from geopy.geocoders.base import Geocoder, DEFAULT_TIMEOUT, DEFAULT_SCHEME
+from geopy.exc import (
+    GeocoderQuotaExceeded,
+    GeocoderQueryError,
+)
+from geopy.location import Location
+
+
+__all__ = ("GooglePlaces", )
+
+
+class GooglePlaces(Geocoder):  # pylint: disable=R0902
+    """
+    Geocoder using the Google Places API. Documentation at:
+        https://developers.google.com/places/documentation/
+    """
+
+    def __init__(
+            self,
+            api_key=None,
+            domain='maps.googleapis.com',
+            scheme=DEFAULT_SCHEME,
+            timeout=DEFAULT_TIMEOUT,
+            proxies=None,
+            user_agent=None
+    ):  # pylint: disable=R0913
+
+        """
+        Initialize a customized Google geocoder.
+
+        API authentication is only required for Google Maps Premier customers.
+
+        :param string api_key: The API key required by Google to perform
+            geocoding requests. API keys are managed through the Google APIs
+            console (https://code.google.com/apis/console).
+
+        :param string domain: Should be the localized Google Maps domain to
+            connect to. The default is 'maps.googleapis.com', but if you're
+            geocoding address in the UK (for example), you may want to set it
+            to 'maps.google.co.uk' to properly bias results.
+
+        :param string scheme: Use 'https' or 'http' as the API URL's scheme.
+            Default is https. Note that SSL connections' certificates are not
+            verified.
+
+        :param dict proxies: If specified, routes this geocoder's requests
+            through the specified proxy. E.g., {"https": "192.0.2.0"}. For
+            more information, see documentation on
+            :class:`urllib2.ProxyHandler`.
+        """
+        super(GooglePlaces, self).__init__(
+            scheme=scheme, timeout=timeout, proxies=proxies, user_agent=user_agent
+        )
+
+        self.api_key = api_key
+        self.domain = domain.strip('/')
+        self.scheme = scheme
+        self.doc = {}
+
+        self.autocomplete_api = '%s://%s/maps/api/place/autocomplete/json' % (self.scheme, self.domain)
+        self.details_api = '%s://%s/maps/api/place/details/json' % (self.scheme, self.domain)
+
+    @staticmethod
+    def _format_components_param(components):
+        """
+        Format the components dict to something Google understands.
+        """
+        return "|".join(
+            (":".join(item)
+             for item in components.items()
+             )
```

Comment:
unless i'm misunderstanding pyhton's `.join` method the `"|".join` will never be called because you're always passing in exactly one element